languages/translations/pt-BR.json

```diff
@@ -112,7 +112,7 @@
                 "upcoming": "Você vai chegar ao seu {nth} destino, em frente",
                 "short": "Você chegou",
                 "short-upcoming": "Você vai chegar",
-                "named": "You have arrived at {waypoint_name}, straight ahead"
+                "named": "Você chegou a (ao) {waypoint_name}, logo em frente"
```

Comment:
The phrase “a (ao)” won’t sound good when read aloud by a text-to-speech engine. Are both prepositions just as likely to occur with a waypoint name? Is there a way we can detect which is more appropriate by matching the waypoint name against regular expressions? If so, we can add [grammar rules](https://github.com/Project-OSRM/osrm-text-instructions/tree/master/languages/grammar/) for Brazilian Portuguese.

Reply:
Yes, adding an article in proper form is not a hard task, please look at [Hungarian rules block](https://github.com/Project-OSRM/osrm-text-instructions/blob/master/languages/grammar/hu.json#L6-L13) for this.

Reply:
Right, I just don’t know what which specific rules we would use. _Ao_ is a contraction meaning “of the”, so I guess it would be similar to the French rules for _l’_.

Reply:
@1ec5 the "ao" is used when the word is in male gender and "a" when is in female gender.
An example:
You have arrived at **Frei** Caneca Shopping > Você chegou **ao Shopping** Frei Caneca
You have arrived at **Escola** Municipal Machado de Assis > Você chegou **a Escola** Municipal Machado de Assis.
We translated in this form to work in this two cases, but of course is better if use only the correct for that case.

Reply:
The destination name could be an arbitrary string, so unfortunately OSRMTI doesn’t automatically know the gender when forming instructions. In some other languages, we’ve created rules that try to detect the gender based on spelling patterns. For example, a regular expression could insert “ao” when the first word in the destination name ends in -o or “a” when it ends in -a or -ção. But it would probably get quite complex due to irregular nouns, determiners, etc.